src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il

```diff
@@ -262,13 +262,19 @@
   .method public hidebysig static !!T& AsRef<T>(void* source) cil managed aggressiveinlining
   {
         .custom instance void System.Runtime.Versioning.NonVersionableAttribute::.ctor() = ( 01 00 00 00 )
+#ifdef NETCORE20
+        .maxstack 1
+        ldarg.0
+        ret
+#else
```

Comment:
Would it make sense to add explanation why we cannot use the if-branch always?
I assume there is some reason like a bug fix in JIT?